File: Python/stack/validate_stack_sequences.py
from typing import *

def validateStackSequences(pushed: List[int], popped: List[int]) -> bool:
    m, n = len(pushed), len(popped)
    if m != n:
        return False
    if m == 0:
        return True 
    stack = list()
    point1, point2 = 0, 0
    while point2 < n and point1 < m:
        stack.append(pushed[point1])
        point1 += 1
        if stack[-1] == popped[point2]:
            while stack and stack[-1] == popped[point2]:
                stack.pop()
                point2 += 1
    
    return len(stack) == 0

File: core/tmp/Java/src/com/dbc/algorithm/Stack/ValidateStackSequences.java
package com.dbc.algorithm.Stack;

import java.util.Stack;

public class ValidateStackSequences {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        if (pushed.length != popped.length) return false;
        if (pushed.length == 0) return true;

        int point1 = 0, point2 = 0;
        Stack<Integer> stack = new Stack<>();
        while (point1 < pushed.length && point2 < popped.length) {
            stack.push(pushed[point1]);
            point1++;
            if (stack.peek() == popped[point2]) {
                while (!stack.isEmpty() && stack.peek() == popped[point2]) {
                    stack.pop();
                    point2++;
                }
            }
        }
        return stack.size() == 0;
    }
}
